rust/toolchain.bzl

```diff
@@ -246,7 +258,15 @@ def _rust_toolchain_impl(ctx):
         binary_ext = ctx.attr.binary_ext,
         staticlib_ext = ctx.attr.staticlib_ext,
         dylib_ext = ctx.attr.dylib_ext,
+        # Contains linker flags needed to link Rust standard library.
+        # These need to be added to linker command lines when the linker is not rustc
+        # (rustc does this automatically). Only use these flags for direct actions and
+        # Use `stdlib_linkflags_cc_info for propagation (otherwise `stdlib_linkflags`
+        # may end up being unnecessailry duplicated).
         stdlib_linkflags = ctx.attr.stdlib_linkflags,
```

Comment:
Kinda surprised this isn't available via [CcInfo](https://docs.bazel.build/versions/4.0.0/skylark/lib/CcInfo.html) somehow (just looking at `compilation_context` and `linking_context`) or [cc_common](https://docs.bazel.build/versions/4.0.0/skylark/lib/cc_common.html). Do you know if there actually is a way to get the linker flags again in Starlark? Not a huge deal if not but I think it'd be nice to allow the `stdlib_linkflags_cc_info` attribute to be the single source of truth.

Reply:
Good idea. You can access the flags via `cc_info.linking_context.linker_inputs[0].user_link_flags`. I feel like people don't rely directly on `ToolchainInfo` from `rust_toolchain`, so I'd just replace the old field with the new one. What do you think?

Reply:
(done in the latest commit)

Reply:
Since it's part of the public API, I think it might be worth having some kind of deprecation notice. Though, I think if you were to add the migration story to the PR description then it wouldn't be the end of the world if anyone decides to pick up this change.


Also, Taking a second look at this, I wonder if it'd be worth just having a `cc_info` field on `rust_toolchain` to allow the toolchain to be more flexible? Would there be additional uses for `CcInfo` aside from housing link flags?

Reply:
Added a note to the PR description.

Re one big `CcInfo` from `rust_toolchain`, that's the vision yes. But I'd leave that to a followup PR as I expect that will need a bit more investigation and added test coverage.

Reply:
Sounds good! Last question just to double check, `cc_info.linking_context.linker_inputs` [is reported to be a `depset`](https://docs.bazel.build/versions/4.0.0/skylark/lib/LinkingContext.html#linker_inputs) which I thought couldn't be indexed. Am I mistaken?

Reply:
Nice catch, yeah you have to `.to_list()` first. Updated the description.